client/lib/site/computed-attributes.js

```diff
@@ -0,0 +1,61 @@
+/** @ssr-ready **/
+
+/**
+ * External dependencies
+ */
+import trim from 'lodash/trim';
+
+/**
+ * Internal dependencies
+ */
+import { isHttps } from 'lib/url';
+import config from 'config';
+
+export default function( site ) {
+	const attributes = {};
+
+	// If the user has no access to site.options create it as an empty
+	// attribute to avoid potential errors when trying to access its sub properties
+	attributes.options = site.options || {};
+
+	// Add URL without protocol as a `domain` attribute
+	if ( site.URL ) {
+		attributes.domain = site.URL.replace( /^https?:\/\//, '' );
+		attributes.slug = attributes.domain.replace( /\//g, '::' );
+	}
+	attributes.title = trim( site.name ) || attributes.domain;
+
+	// If a WordPress.com site has a mapped domain create a `wpcom_url`
+	// attribute to allow site selection with either domain.
+	if ( attributes.options && attributes.options.is_mapped_domain && ! site.is_jetpack ) {
+		attributes.wpcom_url = site.options.unmapped_url.replace( /^https?:\/\//, '' );
+	}
+
+	// If a site has an `is_redirect` property use the `unmapped_url`
+	// for the slug and domain to match the wordpress.com original site.
+	if ( ( attributes.options && attributes.options.is_redirect ) || site.hasConflict ) {
```

Comment:
>It's being set in `sites-list`, in`initialize()` and `update()`

But this attribute will be (rightfully) [stripped from the state object](https://github.com/Automattic/wp-calypso/blob/8527d55850218ce2b8429cc6fe1331e3628c7a85/client/state/sites/reducer.js#L55) when received into the reducer.